tools/pythonpkg/src/pyresult.cpp

```diff
@@ -203,6 +205,22 @@ py::object DuckDBPyResult::Fetchone() {
 	return move(res);
 }
 
+py::list DuckDBPyResult::Fetchmany(idx_t size) {
+	py::list res;
+	for (idx_t i = 0; i < size; i++) {
+		try {
+			auto fres = Fetchone();
+			if (fres.is_none()) {
+				break;
+			}
+			res.append(fres);
+		} catch (InvalidInputException &e) {
```

Comment:
Hmm, quite a broad catch. Is it worth checking the error message?

Reply:
Fixed it to not need a try-catch